src/storage.cpp

```diff
@@ -1692,6 +1693,9 @@ namespace libtorrent
 	{
 		TORRENT_ASSERT(m_files.piece_length() > 0);
 
+		if (checked_fastresume) return check_no_fastresume(ec);
```

Comment:
Let me explain a little more, since this is an important issue for jlibtorrent, we are getting a lot of errors like this:
```
backtrace:
    #00  pc 00254854  /data/app-lib/com.frostwire.android-1/libjlibtorrent.so (libtorrent::bdecode_node::type() const+16)
    #01  pc 0030ccbc  /data/app-lib/com.frostwire.android-1/libjlibtorrent.so (libtorrent::piece_manager::check_fastresume(libtorrent::bdecode_node const&, std::vector<std::string, std::allocator<std::string> > const*, libtorrent::storage_error&)+28)
```
You mentioned in https://github.com/arvidn/libtorrent/issues/781 that probably `async_check_fastresume` was called twice and the first one resulted in an error. Then the second call ends in a crash.

I think this is exactly the reason, since the potential problems of emulating a posix layer with the android SD card framework is just insane.

This just prevents a double inspection of the job `bdecode_node` pointer.

If I restore the logic, my test fail of course, but that is because I expect this logic in place, see `on_checked_resume_data` for `n==2` and `n==3` the `TEST_EQUAL(j->error.ec.value(), 0);`

I rather detect the exact logic flow that creates the two calls to `async_check_fastresume`, but I couldn't.